### PR TITLE
Create pybitmessage.profile

### PR DESCRIPTION
**Pull request by 1dnrr**
tested on fedora-28 with pybitmessage 0.6.3.2

### CODE REVIEW

File: etc/pybitmessage.profile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Firejail profile for default
```
Comment (Vincent43) [inline etc/pybitmessage.profile]: change `default`  for `pybitmessage`

File: etc/pybitmessage.profile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# This file is overwritten after every install/update
+# Persistent local customizations
+include /etc/firejail/pybitmessage.local
+
```
Comment (Vincent43) [inline etc/pybitmessage.profile]: Add:
```
# Persistent global definitions
include /etc/firejail/globals.local

```

File: etc/pybitmessage.profile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# generic gui profile
+# depending on your usage, you can enable some of the commands below:
```
Comment (Vincent43) [inline etc/pybitmessage.profile]: delete those two lines

File: etc/pybitmessage.profile [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+noblacklist /sbin
+noblacklist /usr/local/sbin
+noblacklist /usr/sbin
```
Comment (Vincent43) [inline etc/pybitmessage.profile]: I don't see the reason for those three. Delete them.

File: etc/pybitmessage.profile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+private-etc PyBitmessage,PyBitmessage.conf,Trolltech.conf,fonts,gtk-2.0,hosts,ld.so.cache,ld.so.preload,localtime,pki,resolv.conf,selinux,sni-qt.conf,system-fips,xdg
+#private-lib
+private-tmp
+ca-certificates,ssl,pki,crypto-policies
+
+memory-deny-write-execute
+noexec ${HOME}
+noexec /sbin
+noexec /usr/local/sbin
+noexec /usr/sbin
```
Comment (Vincent43) [inline etc/pybitmessage.profile]: Delete above three noexec rules from sbin

File: etc/pybitmessage.profile (hunk 4 above)
Comment (1dnrr) [reply etc/pybitmessage.profile]: I get the error and crush pybitmessage start if I comment all or any this three lines.
```
Traceback (most recent call last):
  File "bitmessagemain.py", line 19, in <module>
    import depends
  File "/usr/share/pybitmessage/depends.py", line 5, in <module>
    import pyelliptic.openssl
  File "/usr/share/pybitmessage/pyelliptic/__init__.py", line 16, in <module>
    from .openssl import OpenSSL
  File "/usr/share/pybitmessage/pyelliptic/openssl.py", line 549, in <module>
    loadOpenSSL()
  File "/usr/share/pybitmessage/pyelliptic/openssl.py", line 547, in loadOpenSSL
    raise Exception("Couldn't find and load the OpenSSL library. You must install it.")
Exception: Couldn't find and load the OpenSSL library. You must install it.
```
```
/etc/firejail/disable-common.inc:blacklist /sbin
/etc/firejail/disable-common.inc:blacklist /usr/local/sbin
/etc/firejail/disable-common.inc:blacklist /usr/sbin
```

Comment (Vincent43) [reply etc/pybitmessage.profile]: Does it work if you add "openssl*" to private-bin option?

Comment (1dnrr) [reply etc/pybitmessage.profile]: No, I checked it does not work. The same error occurs.

File: etc/pybitmessage.profile [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+seccomp
+shell none
+
+disable-mnt
+private-bin pybitmessage,python*,sh,ldconfig,env,bash,stat
+private-dev
+private-etc PyBitmessage,PyBitmessage.conf,Trolltech.conf,fonts,gtk-2.0,hosts,ld.so.cache,ld.so.preload,localtime,pki,resolv.conf,selinux,sni-qt.conf,system-fips,xdg
+#private-lib
+private-tmp
+ca-certificates,ssl,pki,crypto-policies
```
[the remark this answers is not in the document]
Comment (1dnrr) [reply etc/pybitmessage.profile]: pybitmessage does not work with plain private-etc (without PyBitmessage,PyBitmessage.conf,Trolltech...)
Do you think that the private-etc line should be removed?

File: etc/pybitmessage.profile [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+# This file is overwritten after every install/update
+# Persistent local customizations
+include /etc/firejail/pybitmessage.local
+# Persistent global definitions
+include /etc/firejail/globals.local
+
+
+noblacklist /sbin
+noblacklist /usr/local/sbin
+noblacklist /usr/sbin
```
Comment (Vincent43) [inline etc/pybitmessage.profile]: I'm still interested if noblacklist sbin is needes after noblacklist python was put in the right place

Comment (1dnrr) [reply etc/pybitmessage.profile]: Just checked. Pybitmessage doesn't working with the "noblacklist python" lines and without this "/sbin" lines. These "/sbin" lines are necessary.

Comment (Vincent43) [reply etc/pybitmessage.profile]: ok, thx

**Comment by Vincent43** [comment]
Merged, thank you.

**Comment by PeterSurda** [comment]
Thank you. I'm not a firejail expert, but don't you also need

    mkdir ~/.config/PyBitmessage
    whitelist ${HOME}/.config/PyBitmessage

?

**Comment by Vincent43** [comment]
@1dnrr @PeterSurda 

Those can be added for `$HOME` hardening but someone would have to confirm it doesn't break anything. They would block access to `${HOME}/PyBitmessage` for example. I'm totally unfamiliar with PyBitmessage and I don't know what is needed for it to work. It would be great if you can test those rules and maybe open PR here.

**Pull request merged**